File: 1784-check-if-binary-string-has-at-most-one-segment-of-ones/1784-check-if-binary-string-has-at-most-one-segment-of-ones.cpp
class Solution {
public:
    bool checkOnesSegment(string s) {
        return string::npos == s.find("01");
        
        // regex p("1+0+1+.*");        // Pattern
        // bool m = regex_match(s, p); // Match
        // return !m;
    }
};

File: 1784-check-if-binary-string-has-at-most-one-segment-of-ones/1784-check-if-binary-string-has-at-most-one-segment-of-ones.java
// import java.util.regex.Pattern;
// import java.util.regex.Matcher;

class Solution {
    public boolean checkOnesSegment(String s) {
        return !s.contains("01");
        
        // Pattern p = Pattern.compile("1+0+1+");
        // Matcher m = p.matcher(s);
        // return !m.find();
    }
}